src/popup/utils/utils.ts

```diff
@@ -336,6 +336,10 @@ export const calculateNameClaimFee = (name: string): BigNumber => calculateFee(
   },
 );
 
+export function urlWithParams(url: string, params: Record<string, any>) {
+  return `${url}?${new URLSearchParams(JSON.parse(JSON.stringify(params)))}`;
```

Comment:
1. Why parse & stringify? I don't see a reason for deep merge.
2. This piece of code assumes that the URL has no query string. In my opinion more safe would be to allow passing any URL and manipulate it. Here's a untested proposition:

```suggestion
  const urlObj = new URL(url);
  Object.keys(params).forEach((key) => urlObj.searchParams.set(key, params[key]));
  return urlObj.toString();
```

Reply:
Please skip  pt2 from my previous comment as it will throw error. It's not possible to use `URL` with relative paths. This makes me convinced that this util makes no sense for us. Whenever you want to create query string just do this:
```
const query = (new URLSearchParams({
  <the data>
})).toString();
const fullUrl = `${path}${query}`;
```
No need for util.